### DISCUSSION TITLE
Inline completion not working as expected

### DISCUSSION THREAD

**Discussion by Pranith12250** [microsoft/vscode-discussions#1107] [Extension Development QnA] [answered]
I am trying to understand how the inline code completion feature of vs code works and how I can use it in an extension. So, to understand its working, I went and looked at its code sample [here](https://github.com/microsoft/vscode-extension-samples/tree/main/inline-completions).
However, as soon as I open and run the code without making any changes at all, I see the following errors:

`Activating extension 'vscode-samples.inline-completion-sample' failed: Extension 'vscode-samples.inline-completion-sample' CANNOT use API proposal: inlineCompletionsAdditions. Its package.json#enabledApiProposals-property declares: but NOT inlineCompletionsAdditions. The missing proposal MUST be added and you must start in extension development mode or use the following command line switch: --enable-proposed-api vscode-samples.inline-completion-sample.`

<img width="451" alt="Screenshot 2024-03-23 at 2 37 41 AM" src="https://github.com/microsoft/vscode-discussions/assets/127399614/ccc72917-3755-4efd-8abf-6c2df1884b9c">

`[vscode-samples.inline-completion-sample]: editor/inlineCompletions/actions is a proposed menu identifier. It requires 'package.json#enabledApiProposals: ["inlineCompletionsAdditions"]' and is only available when running out of dev or with the following command line switch: --enable-proposed-api vscode-samples.inline-completion-sample`

<img width="451" alt="Screenshot 2024-03-23 at 2 38 12 AM" src="https://github.com/microsoft/vscode-discussions/assets/127399614/e9d07dea-9ff6-4d7b-b9d4-54ca964bb2d7">

I try running it regardless to see if it still works and it opens a new window as a playground with the below code:

```
// Trigger inline completions after each line

// [0,*):// Trigger inline completions in this line, they work!
// Trigger inline completions in this line

// [0,*):console.log(


// [0,*):console.log("foo"
// [0,*):console.log({ label: "("


// [0,*):console.log(`${(1+2}`)

// [0,*):({\n){

// [33,33):lambda x: x.notnull()
notNull = languages.apply();

```

I try typing in "console" just as they did in the demo gif in the sample code page I linked earlier but I am not getting the inline completion. I am just getting normal completion:

<img width="932" alt="Screenshot 2024-03-23 at 2 24 20 AM" src="https://github.com/microsoft/vscode-discussions/assets/127399614/25e2bfff-7adc-42f5-a397-f40e8967f0ef">

Also, I noticed that gif in the sample code is slightly cropped so I am not fully sure how they activated the inline completion as well. So, I need help with two things:

1. What are the errors I mentioned earlier and how do I resolve them? and
2. What do I need to do to see the inline completion working in the playground? Is there something that I am not doing right because of which it is not working or is it because of the error I mentioned earlier?

**Comment by tamuratak** [comment, marked as answer]
The sample needs updates. See 

- microsoft/vscode-extension-samples/issues/867
- microsoft/vscode-extension-samples/pull/879

A simple [workaround](https://github.com/microsoft/vscode-extension-samples/issues/867#issuecomment-1890975137) is to install [VS Code Insiders](https://code.visualstudio.com/insiders/) .
